corehq/form_processor/utils/general.py

```diff
@@ -9,3 +12,35 @@ def should_use_sql_backend(domain):
     # uncomment when ready for production
     # return USE_SQL_BACKEND.enabled(domain)
     return False
+
+
+def cache_return_value(fn):
+    """
+    Docorator to save the return value of a method or property.
+    Use this where you can't use quickcache or memoize because you can't
+    guarantee a consistent or unique lookup key e.g. forms that get deprecated
+
+    Requires that the first arg is 'self' and that no other args or kwargs are being
+    passed in.
+
+    :param fn: a function or property with no args (other than 'self')
+    """
```

Comment:
is the idea here that functions that have no args will always return the same value? i think one major caveat could be if it's looking up properties on the model itself. if those have changed than the output of the function will change. if so it seems like this could be a big dangerous and we might want to put that caveat in the doc string